packages/mdc-shape/_functions.scss

```diff
@@ -114,15 +87,27 @@
       @error "Invalid radius: '#{$radius}' is more than 4 values";
     }
 
-    $radius-value: ();
+    $radius-values: ();
 
-    @each $corner in $radius {
-      $radius-value: append($radius-value, mdc-shape-prop-corner-value_($corner));
+    @for $i from 1 through length($radius) {
+      $corner: nth($radius, $i);
+
+      @if map-has-key($mdc-shape-category-values, $corner) {
+        // If a category is encountered within a list of radii, apply the category's value for the corresponding corner
```

Comment:
This behavior maximizes compatibility with the existing behavior which clearly expected shape categories to be applicable to individual corners (e.g. `small small 0 0`). This will effectively behave identically to the existing behavior in cases where a shape category is a single radius, while making list-of-radii category values actually functional.